src/SmartComponents/BaselinesPage/EditBaseline/EditBaseline.js

```diff
@@ -1,128 +1,410 @@
+                            autoFocus={ activeEditId === id }
+                        />
+                    );
+                }) }
+            </ExpandableRowContent>
+        );
+
+        const parentEditRenderer = (value, { column, rowIndex, columnIndex, activeEditId }) => {
+            const id = this.makeId({ rowIndex, columnIndex, column, name: 'parent-repo' });
+            return (
+                <TableTextInput
+                    id={ id }
+                    key={ id }
+                    defaultValue={ value }
+                    onBlur={ newValue =>
+                        this.onChange(newValue, {
+                            rowIndex,
+                            columnIndex
+                        })
+                    }
+                    autoFocus={ activeEditId === id }
+                />
+            );
+        };
 
-    renderRows() {
-        const { baselineData } = this.props;
-        let rows = [];
+        const textInputFormatter = inlineEditFormatterFactory({
+            renderValue: (value, { rowData }) =>
+                rowData.hasOwnProperty('parent') ? (
+                    <ExpandableRowContent>{ rowData.data.modules.filter(a => a).join(', ') }</ExpandableRowContent>
+                ) : (
+                    value
+                ),
+            renderEdit: (value, { column, columnIndex, rowIndex, rowData }, { activeEditId }) => {
+                const renderer = rowData.hasOwnProperty('parent') ? childEditRenderer : parentEditRenderer;
+                return renderer(value, { rowData, column, rowIndex, columnIndex, activeEditId });
+            }
+        });
 
-        if (!baselineData) {
-            rows = [];
+        /*eslint-disable react/jsx-key*/
+        let rows = [ ...new Array(5) ].map(() =>
+            [ ...new Array(2) ].map(() =>
+                <Skeleton size={ SkeletonSize.md } />
+            )
+        );
+        /*eslint-enable react/jsx-key*/
+
+        let displayName;
+
+        if (this.props.baselineData) {
+            displayName = this.props.baselineData.display_name;
         } else {
-            baselineData.baseline_facts.forEach(function(fact) {
-                let row = [];
-                row.push(fact.name);
-                row.push(fact.value);
-                rows.push(row);
-            });
+            displayName = '';
         }
 
-        return rows;
-    }
+        this.state = {
+            columns: [ 'Fact', 'Value' ],
+            rows,
+            editedRowsBackup: null,
+            activeEditId: null,
+            baselineName: displayName
+        };
 
-    renderLoadingRows() {
-        let rows = [];
-        let rowData = [];
+        this.changeBaselineName = value => {
+            this.setState({ value });
+        };
 
-        for (let i = 0; i < 2; i += 1) {
-            rowData.push(<Skeleton size={ SkeletonSize.md } />);
-        }
+        this.submitBaselineName = (/*baselineName*/) => {
```

Comment:
dunno if this inline comment is intentional or not

Reply:
I'm leaving it there for when I do the next phase of this.